api/cpp/include/slint_platform.h

```diff
@@ -178,8 +221,26 @@ class Platform
     /// This function should only be implemented  if the runtime is compiled with no_std
     virtual uint64_t duration_since_start() const { return 0; }
 
-    /// Register the platform to Slint. Must be called before Slint window are created. Can only
-    /// be called once in an application.
+    /// Spins an event loop and renders the visible windows.
+    virtual void run_event_loop() const { }
+
+    /// Exits the event loop.
+    ///
+    /// This is what is called by slint::quit_event_loop() and can be called from a different thread
+    /// or re-enter from the event loop
+    virtual void quit_event_loop() const { }
+
+    /// Invokes the event from the event loop.
+    ///
+    /// This function is called by slint::invoke_from_event_loop().
+    /// It can be called from any thread, but the passed function must only be called
+    /// from the event loop.
+    /// Reimplements this function and move the event to the event loop before calling
+    /// PlatformEvent::invoke
+    virtual void invoke_from_event_loop(PlatformEvent) const { }
```

Comment:
At first I thought that your initial draft of taking an std::function is a nicer API, but I realise it's only _us_ calling this function, and we don't need a nice API. We need to make it easy for implementors of this function, and for that the PlatformEvent you propose makes sense.